concent_api/core/views.py

```diff
@@ -1,223 +1,321 @@
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key
+        )
+        if message_task_to_compute.deadline + settings.CONCENT_MESSAGING_TIME <= current_time:
+            message_verdict.message_force_report_computed_task = dump(
+                decoded_last_task_message,
+                settings.CONCENT_PRIVATE_KEY,
+                client_public_key,
+            )
+            message_ack_report_computed_task = MessageAckReportComputedTask(
+                message_task_to_compute = decoded_last_task_message.message_task_to_compute
+            )
+            message_verdict.message_ack_report_computed_task = dump(
+                message_ack_report_computed_task,
+                settings.CONCENT_PRIVATE_KEY,
+                client_public_key,
+            )
             return message_verdict
 
-    if decoded_last_task_message['type'] == "MessageRejectReportComputedTask":
-        if decoded_last_task_message['message_cannot_commpute_task']['reason'] == "deadline-exceeded":
-            rejected_task_id                                      = decoded_last_task_message['message_cannot_commpute_task']['task_id']
-            message_to_compute                                    = Message.objects.get(type = 'MessageForceReportComputedTask', task_id = rejected_task_id)
-            raw_message_to_compute                                = message_to_compute.data.tobytes()
-            decoded_message_to_compute                            = json.loads(raw_message_to_compute.decode('utf-8'))
-            message_verdict['message_force_report_computed_task'] = decoded_message_to_compute
+    if isinstance(decoded_last_task_message, MessageRejectReportComputedTask):
+        message_cannot_compute_task = load(
+            decoded_last_task_message.message_cannot_compute_task,
+            settings.CONCENT_PRIVATE_KEY,
+            client_public_key
+        )
+        if decoded_last_task_message.reason == "deadline-exceeded":
+            rejected_task_id           = message_cannot_compute_task.task_id
+            message_to_compute         = Message.objects.get(type = 'MessageForceReportComputedTask', task_id = rejected_task_id)
+            raw_message_to_compute     = message_to_compute.data.tobytes()
+            force_report_computed_task = load(
+                raw_message_to_compute,
+                settings.CONCENT_PRIVATE_KEY,
+                client_public_key
+            )
+            message_ack_report_computed_task.message_task_to_compute    = force_report_computed_task.message_task_to_compute
+            dumped_message_ack_report_computed_task = dump(
+                message_ack_report_computed_task,
+                settings.CONCENT_PRIVATE_KEY,
+                client_public_key
+            )
+            message_verdict.message_ack_report_computed_task = dumped_message_ack_report_computed_task
             return message_verdict
 
     return HttpResponse("", status = 204)
```

Comment:
Same here. You cannot simply return if the current message was not returned. You have to check if there are any other messages waiting to be delivered. If so, process them until you find one that can be delivered or you run out of messages.